mobile/lib/shared/providers/release_info.provider.dart

```diff
@@ -23,35 +41,70 @@ class ReleaseInfoNotifier extends StateNotifier<String> {
         headers: {"Accept": "application/vnd.github.v3+json"},
       );
 
+      ServerVersionResponseDto? serverVersion =
+          await _serverInfoService.getServerVersion();
+
       if (res.statusCode == 200) {
         final data = jsonDecode(res.body);
-        String latestTagVersion = data["tag_name"];
-        state = latestTagVersion;
+        String latestTagVersion = data["tag_name"].replaceAll("v", "");
+        List<String> latestTagVersionSectioned = latestTagVersion.split(".");
+        state = state.copyWith(
+          serverVersion: ServerVersionResponseDto(
+            major: latestTagVersionSectioned[0].toInt(),
+            patch_: latestTagVersionSectioned[2].toInt(),
+            minor: latestTagVersionSectioned[1].toInt(),
+          ),
+        );
+
+        // If server is already updated, ignore the new version announcement
+        if (serverVersion != null &&
+            Version.parse(serverVersionReponseToString(serverVersion)) ==
+                Version.parse(latestTagVersion)) {
+          debugPrint("Server is already updated");
+          return;
+        }
 
         if (localReleaseVersion == null && latestTagVersion.isNotEmpty) {
           VersionAnnouncementOverlayController.appLoader.show();
+          debugPrint("Release accouncement has not been shown yet");
           return;
         }
 
+        // If local release version is not the same as the latest release version, show the new version announcement
         if (latestTagVersion.isNotEmpty &&
-            localReleaseVersion != latestTagVersion) {
+            (localReleaseVersion == null ||
+                Version.parse(localReleaseVersion.replaceAll("v", "")) <
+                    Version.parse(latestTagVersion))) {
           VersionAnnouncementOverlayController.appLoader.show();
           return;
         }
       }
     } catch (e) {
       debugPrint("Error gettting latest release version");
 
-      state = "";
+      state = ServerInfoState(
+        serverVersion: ServerVersionResponseDto(
+          major: 0,
+          patch_: 0,
+          minor: 0,
+        ),
+        isVersionMismatch: false,
+        versionMismatchErrorMessage: "",
+      );
     }
   }
 
+  String serverVersionReponseToString(ServerVersionResponseDto dto) {
```

Comment:
Can you refactor this method to a common util file to now have duplicated code in two places? also there is typo in the function name

Reply:
Sorry about that.. Done!